introspection-engine/connectors/sql-introspection-connector/src/introspection_helpers.rs

```diff
@@ -347,8 +353,14 @@ pub(crate) fn calculate_relation_name(schema: &SqlSchema, fk: &ForeignKey, table
                 .iter()
                 .any(|fk| &fk.referenced_table == model_with_fk);
 
-            let name = if fk_to_same_model.len() < 2 && !fk_from_other_model_to_this_exist {
-                RelationNames::name_for_unambiguous_relation(model_with_fk, referenced_model)
+            let unambiguous_name = RelationNames::name_for_unambiguous_relation(model_with_fk, referenced_model);
```

Comment:
Why are we creating this variable here, not in the first if block a bit down from here?

Reply:
Because its used in the if condition before getting to the first block